server/text_generation_server/layers/fp8.py

```diff
@@ -276,6 +296,22 @@ def get_multi_weights_col(self, weights: "Weights", prefixes: List[str], dim: in
 
         # FP8 branch
         if w.dtype == torch.float8_e4m3fn:
+            if self.weight_block_size is not None:
+                scale = [
+                    weights.get_sharded(f"{p}.weight_scale_inv", dim=0, to_device=False)
+                    for p in prefixes
+                ]
+                scale = torch.cat(scale, dim=dim)
+                if scale.device == torch.device("cpu"):
```

Comment:
When does this happen?

Reply:
I was testing something here: but I guess we can do: 

```python
 scale = torch.cat(scale, dim=dim).to(weights.device)
```